konfera/views.py

```diff
@@ -2,14 +2,15 @@
 from konfera.models import Event, Sponsor
 
 
-def sponsor_list_view(request, event_id):
+def sponsor_list_view(request, event_slug):
+    event = get_object_or_404(Event, slug=event_slug)
     sponsors = Sponsor.objects.select_related('event') \
-                    .filter(event__id=event_id).order_by('type')
-    event = Event.objects.get(pk=event_id)
+                    .filter(event=event).order_by('type')
+
     context = {'event': event,
-               'sponsors': sponsors,
-               }
+               'sponsors': sponsors, }
+    template = 'konfera/event_sponsors.html'
```

Comment:
I dont think it is necessary to introduce new variable here.

It not used anywhere else in the view, so I think original setup (pass string as a parameter or render) was ok, please revert back.